Cannot leave LS_ACCESS_TOKEN empty when using Developer Satellite

According to [Lightstep Documentation](https://docs.lightstep.com/docs/quick-start-1), when using a Developer Satellite, you should be able to leave the `LS_ACCESS_TOKEN` field blank. However, when this field is blank, the following error is logged, and traces do not appear to be sent:

```
2021-03-31T23:24:32.286895300Z Picked up JAVA_TOOL_OPTIONS: -javaagent:/tools/lightstep-opentelemetry-javaagent.jar
2021-03-31T23:24:32.601762000Z [opentelemetry.auto.trace 2021-03-31 23:24:32:601 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 0.16.0
2021-03-31T23:24:32.654755700Z [opentelemetry.auto.trace 2021-03-31 23:24:32:654 +0000] [main] ERROR io.opentelemetry.javaagent.bootstrap.AgentInitializer - Throwable thrown while installing the agent
2021-03-31T23:24:32.654893600Z java.lang.reflect.InvocationTargetException
2021-03-31T23:24:32.655239900Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-03-31T23:24:32.655250300Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-03-31T23:24:32.655254500Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-03-31T23:24:32.655341700Z    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-03-31T23:24:32.655352300Z    at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:65)
2021-03-31T23:24:32.655354800Z    at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:51)
2021-03-31T23:24:32.655477700Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-03-31T23:24:32.655487300Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-03-31T23:24:32.655602100Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-03-31T23:24:32.655612100Z    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-03-31T23:24:32.655615500Z    at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
2021-03-31T23:24:32.655696800Z    at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
2021-03-31T23:24:32.655811500Z    at com.lightstep.opentelemetry.agent.LightstepBootstrap.premain(LightstepBootstrap.java:18)
2021-03-31T23:24:32.655823500Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-03-31T23:24:32.655905600Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-03-31T23:24:32.656016200Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-03-31T23:24:32.656025600Z    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-03-31T23:24:32.656129000Z    at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
2021-03-31T23:24:32.656136200Z    at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
2021-03-31T23:24:32.656372200Z Caused by: io.opentelemetry.sdk.autoconfigure.ConfigurationException: Invalid map property: otel.exporter.otlp.headers=lightstep-access-token=
2021-03-31T23:24:32.656379400Z    at io.opentelemetry.sdk.autoconfigure.ConfigProperties.lambda$getCommaSeparatedMap$3(ConfigProperties.java:198)
2021-03-31T23:24:32.656390900Z    at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
2021-03-31T23:24:32.656411500Z    at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
2021-03-31T23:24:32.656590400Z    at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
2021-03-31T23:24:32.656599800Z    at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
2021-03-31T23:24:32.656616300Z    at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
2021-03-31T23:24:32.656624200Z    at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
2021-03-31T23:24:32.656690300Z    at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2021-03-31T23:24:32.656702200Z    at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
2021-03-31T23:24:32.656768500Z    at io.opentelemetry.sdk.autoconfigure.ConfigProperties.getCommaSeparatedMap(ConfigProperties.java:205)
2021-03-31T23:24:32.656916900Z    at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureOtlpSpans(SpanExporterConfiguration.java:75)
2021-03-31T23:24:32.656927900Z    at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureExporter(SpanExporterConfiguration.java:42)
2021-03-31T23:24:32.657003800Z    at io.opentelemetry.sdk.autoconfigure.TracerProviderConfiguration.configureTracerProvider(TracerProviderConfiguration.java:43)
2021-03-31T23:24:32.657012200Z    at io.opentelemetry.sdk.autoconfigure.OpenTelemetrySdkAutoConfiguration.initialize(OpenTelemetrySdkAutoConfiguration.java:34)
2021-03-31T23:24:32.657015400Z    at io.opentelemetry.javaagent.tooling.OpenTelemetryInstaller.installAgentTracer(OpenTelemetryInstaller.java:36)
2021-03-31T23:24:32.657133700Z    at io.opentelemetry.javaagent.tooling.OpenTelemetryInstaller.beforeByteBuddyAgent(OpenTelemetryInstaller.java:27)
2021-03-31T23:24:32.657250500Z    at io.opentelemetry.javaagent.tooling.AgentInstaller.installComponentsBeforeByteBuddy(AgentInstaller.java:164)
2021-03-31T23:24:32.657259700Z    at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:98)
2021-03-31T23:24:32.657261900Z    at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:82)
2021-03-31T23:24:32.657355500Z    ... 19 more
```

I believe the issue is that [this system property](https://github.com/lightstep/otel-launcher-java/blob/eaa190e97365070421e4d1f4d23cfed87604b398/common/src/main/java/com/lightstep/opentelemetry/common/VariablesConverter.java#L98-L99) is malformed when the access token is missing.

There is a simple workaround - just providing a valid Lightstep access token, which is ignored by the Developer Satellite. Still, it'd be nice if this was made optional, since access token shouldn't be required for the Developer Satellite to work.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 <a name="Pending Release"></a>
 ## [Pending Release](https://github.com/lightstep/otel-launcher-java/compare/1.0.1...master)
+* Allow empty LS_ACCESS_TOKEN when using Developer Satellite
 
 <a name="1.0.1"></a>
 ## [1.0.1](https://github.com/lightstep/otel-launcher-java/compare/1.0.0...1.0.1)

--- a/common/src/main/java/com/lightstep/opentelemetry/common/VariablesConverter.java
+++ b/common/src/main/java/com/lightstep/opentelemetry/common/VariablesConverter.java
@@ -95,8 +95,12 @@ public class VariablesConverter {
     System.setProperty("otel.exporter.otlp.traces.endpoint", configuration.tracesEndpoint);
     System
         .setProperty("otel.exporter.otlp.timeout", String.valueOf(DEFAULT_LS_DEADLINE_MILLIS));
-    System.setProperty("otel.exporter.otlp.headers",
-        "lightstep-access-token=" + configuration.accessToken);
+
+    if (configuration.accessToken != null && !configuration.accessToken.isEmpty()) {
+      System.setProperty("otel.exporter.otlp.headers",
+          "lightstep-access-token=" + configuration.accessToken);
+    }
+
     if (configuration.propagators != null) {
       System.setProperty("otel.propagators", configuration.propagators);
     }
@@ -191,7 +195,7 @@ public class VariablesConverter {
   }
 
   public static String getAccessToken() {
-    return getProperty(LS_ACCESS_TOKEN, "");
+    return getProperty(LS_ACCESS_TOKEN, null);
   }
 
   public static String getServiceName() {

--- a/common/src/test/java/com/lightstep/opentelemetry/common/VariablesConverterTest.java
+++ b/common/src/test/java/com/lightstep/opentelemetry/common/VariablesConverterTest.java
@@ -178,7 +178,7 @@ public class VariablesConverterTest {
 
   @Test
   public void getAccessToken_Default() {
-    assertEquals("", VariablesConverter.getAccessToken());
+    assertNull(VariablesConverter.getAccessToken());
   }
 
   @Test

